File: core/src/ru/myitschool/cubegame/effects/ImmobilizedEffect.java
package ru.myitschool.cubegame.effects;

import com.badlogic.gdx.graphics.Texture;
import ru.myitschool.cubegame.entities.Entity;

/**
 * Created by Voyager on 02.12.2017.
 */
public class ImmobilizedEffect extends Effect {

    private String name = "Immobilized";
    private String description = "";
    private Texture icon = new Texture("immobilization.png");
    private boolean positive = false;

    public ImmobilizedEffect(Entity entity, int turns) {
        super(entity);
        setName(name);
        description += "You can not move for " + turns + " turns";
        setDescription(description);
        setIcon(icon);
        setPositive(positive);
        setStackable(false);
        setStackSize(0);
        setExpiring(true);
        setExpireTurns(turns);
        type = 8;
        entity.setImmobilized(true);
    }

    @Override
    public void onExpire() {
        super.onExpire();
        getEntity().setImmobilized(false);
    }
}

File: core/src/com/wolg_vlad/dcrawler/effects/ImmobilizedEffect.kt
package com.wolg_vlad.dcrawler.effects

import com.badlogic.gdx.graphics.Texture
import com.wolg_vlad.dcrawler.entities.Entity

/**
 * Created by Voyager on 02.12.2017.
 */
class ImmobilizedEffect(entity: Entity?, turns: Int) : Effect(entity) {
    override val name = "Immobilized"
    override val description = "You can not move for $turns turn(-s)"
    override val icon = Texture("immobilization.png")
    override val positive: Boolean = false
    override val skillUse: Boolean = false
    override val stackable: Boolean = false
    override val stackSize: Int = 0
    override val expiring: Boolean = true
    override val expireTurns: Int = turns
    override val id: String = "main.dcrawler.effect.immobilized"

    override fun onExpire() {
        super.onExpire()
        entity!!.isImmobilized = false
    }

    init {
        entity!!.isImmobilized = true
    }
}